boards: st: fix twister test failure for stm32u083c_dk

arifbalik opened the pull request:
Test `drivers.gpio.1pin.aw9523b` fails on `stm32u083c_dk` platform because `arduino_i2c` definition does not exists in the device tree of `stm32u083c_dk` board

For the error see; https://github.com/zephyrproject-rtos/zephyr/actions/runs/12270889177/job/34236994087?pr=81187

soburi commented on boards/st/stm32u083c_dk/arduino_r3_connector.dtsi:
```diff
@@ -36,3 +36,4 @@
 };
 
 arduino_serial: &usart2 {};
+arduino_i2c: &i2c1 {};
```
Please also add `arduino_i2c` to `supported:` section of stm32u083c_dk.yaml.yaml file.

arifbalik committed:
boards: st: fix twister test failure for stm32u083c_dk

Test "drivers.gpio.1pin.aw9523b" fails on stm32u083c_dk platform
because "arduino_i2c" definition does not exists in the device tree
of stm32u083c_dk board. Also updated the baord yaml

Signed-off-by: Arif Balik <arifbalik@outlook.com>

erwango commented:
Blocked by https://github.com/zephyrproject-rtos/zephyr/issues/82344 (fix was reverted)
